servers/apps/test/malloc_test.c

```diff
@@ -0,0 +1,43 @@
+#include <resea/malloc.h>
+#include <resea/printf.h>
+#include <string.h>
+#include "test.h"
+#define NUM_PTRS 16
+
+void malloc_test(void) {
+    malloc_init();
```

Comment:
You don't need to call `malloc_init`. It is already called before `main()`.

Reply:
This is why the strange bug you're facing occurs!

Reply:
Ah, I didn't know that. This reduced the number of failing tests!